sys/net/application_layer/dhcpv6/client.c

```diff
@@ -254,7 +254,7 @@ static inline size_t _compose_mud_url_opt(dhcpv6_opt_mud_url_t *mud_url_opt,
 
     mud_url_opt->type = byteorder_htons(DHCPV6_OPT_MUD_URL);
     mud_url_opt->len = byteorder_htons(len);
-    strncpy(mud_url_opt->mudString, mud_url);
+    strncpy(mud_url_opt->mudString, mud_url, len);
```

Comment:
`strncpy` must use the size of the *destination* buffer, otherwise it's a rather pointless exercise.

Reply:
Oh, sorry, you are right! (I have to admit that I am that used to work with buffers.) Could you give me a hint how to check if there is enough space left in the buffer? Would we just call the `abort()` function if this check fails? (Sorry again if I am asking obvious questions.)

Reply:
Unfortunately none of the other functions that assemble the request check the size of the buffer :disappointed: 
At least `_add_ia_pd_from_config()` should also be updated with a parameter of the remaining buffer size, so it doesn't silently overflow the request buffer if the MUD URL is long. (should be a separate commit though)

The dhcpv6 client could need some love and cleanup in general

Reply:
I agree, there is indeed room for improvement! Would you say that this deserves the opening of an issue?

I added a similar check as you suggested before to the `_add_ia_pd_from_config` function in commit 0299a36 (I hope the commit message is appropriate). Right now, no space in the buffer also causes a crash in this function.

Reply:
> I agree, there is indeed room for improvement! Would you say that this deserves the opening of an issue?

Not sure if an issue would help, I'd rather have a pull request :wink:

Reply:
Okay, I agree on that ;)